libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -155,49 +159,107 @@ AC_Fence::AC_Fence()
 #endif
     _singleton = this;
     AP_Param::setup_object_defaults(this, var_info);
+    if (_enabled) {
+        _enabled_fences = _present_fences.get() & ~AC_FENCE_TYPE_ALT_MIN;
+    }
+}
+
+// get a user-friendly list of fences
+uint8_t AC_Fence::get_fence_names(uint8_t fences, char* msg, uint8_t len)
+{
+    if (!fences) {
+        return 0 ;
+    }
+    static const char* FENCE_NAMES[] = {
+        "Max Alt",
+        "Circle",
+        "Polygon",
+        "Min Alt",
+    };
+    uint8_t i = 0, nwritten = 0;;
+    uint8_t nfences = 0;
+    while (fences !=0) {
+        if (fences & 0x1) {
+            uint8_t msglen = 0;
+            if (nfences > 0) {
+                if (!(fences & ~1U)) {
+                    msglen = hal.util->snprintf(msg, len, " and ");
+                } else {
+                    msglen = hal.util->snprintf(msg, len, ", ");
+                }
+            }
+            msg += msglen;
+            len -= msglen;
+            nwritten += msglen;
```

Comment:
```suggestion
                msg += msglen;
                len -= msglen;
                nwritten += msglen;
            }
```
This printing seems overly complex, there must be a easier way.

Reply:
I was thinking ExpandingString might be an alternative

Reply:
Fixed